ddr/lib/ddr-scanner/dwarf/DwarfScanner.cpp

```diff
@@ -209,6 +209,12 @@ DwarfScanner::blackListedDie(Dwarf_Die die, bool *dieBlackListed)
 	bool blacklistedFile = true;
 	bool blacklistedType = true;
 
+/* Since AIX does not provide a DW_AT_comp_dir attribute, so we will not have
+ * a file list to look through, therefore don't blacklist based on file name
+ */
+#if defined(AIXPPC)
+	blacklistedFile = false;
```

Comment:
Isn't this change redundant? If there's no `DW_AT_decl_file` attribute, the outcome should be the same.

Reply:
We set the error status to `DDR_RC_ERROR` and exit the function if the attribute isn't found. 

Looking at the function again, it would probably be better to just remove the throw error part:

https://github.com/mikezhang1234567890/omr/blob/21386ad081a48644c241725a7689dcf706a00aa5/ddr/lib/ddr-scanner/dwarf/DwarfScanner.cpp#L235-L236

Reply:
The call to `dwarf_hasattr` should not fail, but it might say the attribute is not present.

Reply:
I was wrong there. The `DW_AT_comp_dir` is the attribute that's missing, so it fails the check on line 255: `if (declFile <= (Dwarf_Unsigned)_fileNameCount)`